runtime/query.go

```diff
@@ -87,6 +88,18 @@ func (r *Runtime) Query(ctx context.Context, instanceID string, query Query, pri
 			return fmt.Errorf("query dependency %q not found", dep)
 		}
 		depKeys[i] = entry.Name + ":" + entry.RefreshedOn.String()
+		// if catalog entry is a metrics view and it has a policy, we should add user attributes to the cache key
+		if entry.IsMetricsView() {
+			policy := entry.GetMetricsView().Policy
+			if policy != nil {
+				attr := auth.GetClaims(ctx).Attributes()
+				attrKey := ""
+				for k, v := range attr {
+					attrKey += fmt.Sprintf(":%v:%v", k, v)
+				}
+				depKeys[i] += attrKey
+			}
+		}
```

Comment:
Doesn't seem quite right, some issues:

1. Iterating over the attributes has non-deterministic order
2. This section is for dependency names, not for the policy (if something had a dependency on 2 metrics views, the attributes would be added twice, though nothing does that now)
3. It will calculate an attribute-based cache key redundantly since the same already happens in the policy engine. Should such a hash calculation be moved to the claims object and cached there?
4. Since the policy filter relates to how the query is executed, would it make more sense to cache based on the resolved filter instead of the input attributes? The resolved filter could maybe be incorporated into `query.Key()`? 
  - This would enable cache sharing among people who resolve to the same filter, for example if the policy only checks domain.